assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -126,15 +126,19 @@ function MediaProvider({ children }) {
   );
 
   const processor = useCallback(
-    ({ mimeType, posterId, id, src, local }) => {
-      const process = async () => {
+    (el) => {
+      if (!el) {
+        return;
+      }
+      const { mimeType, posterId, id, src, local } = el;
+      const process = () => {
         if (
           allowedVideoMimeTypes.includes(mimeType) &&
           !local &&
           !posterId &&
           id
         ) {
-          await uploadVideoPoster(id, src);
+          uploadVideoPoster(id, src);
```

Comment:
If we don't await this promise here, does it not run?

Reply:
I removed this one because uploadVideoPoster() was not async as far as I can tell. If you look above, the callback isn't an asynchronous function and hence doesn't return a promise. (It does create an asynchronous function inside the callback and then call it, but it doesn't await on the response of that promise).

Hence I thought putting await here wasn't doing anything.

Reply:
On second thought, I decided to change this back to the previous version (with an async await) since I'd prefer to keep this PR related to just the File drag and drop crash issue.